modules/tinymce/src/plugins/directionality/test/ts/browser/DirectionStyleTest.ts

```diff
@@ -0,0 +1,271 @@
+import { context, describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyHooks, TinySelections, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/directionality/Plugin';
+
+type Dir = 'rtl' | 'ltr';
+
+describe('browser.tinymce.plugins.directionality.DirectionStyleTest', () => {
+  const applyDir = (editor: Editor, dir: Dir) =>
+    TinyUiActions.clickOnToolbar(editor, dir === 'ltr' ? 'button[title="Left to right"]' : 'button[title="Right to left"]');
+
+  const testDirectionStyle = (editor: Editor, content: string, dir: Dir, expected: string) => {
+    editor.setContent(content);
+    TinySelections.setCursor(editor, [ 0, 0 ], 0);
+    applyDir(editor, dir);
+    TinyAssertions.assertContent(editor, expected);
+  };
+
+  const baseSettings = {
+    plugins: 'directionality',
+    toolbar: 'ltr rtl',
+    base_url: '/project/tinymce/js/tinymce'
+  };
+
+  context('Inline direction style', () => {
+    const hook = TinyHooks.bddSetup<Editor>(baseSettings, [ Plugin ]);
+
+    it('TINY-9314: Applying ltr with dir="rtl" and direction: rtl', () => testDirectionStyle(
+      hook.editor(),
+      '<p dir="rtl" style="direction: rtl;">Lorem ipsum</p>',
+      'ltr',
+      '<p style="direction: ltr;">Lorem ipsum</p>'
+    ));
+
+    it('TINY-9314: Applying rtl with dir="ltr" and direction: ltr', () => testDirectionStyle(
+      hook.editor(),
+      '<p dir="ltr" style="direction: ltr;">Lorem ipsum</p>',
+      'rtl',
+      '<p dir="rtl" style="direction: rtl;">Lorem ipsum</p>'
```

Comment:
Seems a little clunky to have both a style and a dir attribute basically double information. If we set the style we should unset the attribute since it has lower presence normally but I guess we could detect that make a smart setDir that checks if it needs to set the attribute, style by checking the runtime style value after setting either and unsetting the other.

Reply:
@spocke Yes I agree, the aim of the plugin is to keep the attributes/styles the minimal it needs to be, so I will make the change in my implementation to remove the `direction` style property if it is not needed.